### ISSUE TITLE
Static analysis: dealii-git/include/deal.II/lac/precondition.h

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#3363]
```
dealii-git/include/deal.II/lac/precondition.h	2082	warn	V560 A part of conditional expression is always true.
dealii-git/include/deal.II/lac/precondition.h	2084	warn	V560 A part of conditional expression is always true.
dealii-git/include/deal.II/lac/precondition.h	2081	warn	V560 A part of conditional expression is always true.
dealii-git/include/deal.II/lac/precondition.h	1955	warn	V730 Not all members of a class are initialized inside the constructor. Consider inspecting: theta, delta.
dealii-git/include/deal.II/lac/precondition.h	1934	err	V730 Not all members of a class are initialized inside the constructor. Consider inspecting: matrix_diagonal_inverse.
```

We should address these warnings and errors from the static analysis tool PVS. In response to #3342.

**Comment by bangerth** [comment]
This is in the `PreconditionChebychev` class. The first three warnings are due to template arguments having a specific type and can be ignore. I don't think I understand the class well enough to say anything about the other two.

@kronbichler -- would you mind taking a look?

**Comment by kronbichler** [comment]
The first three warnings are due to the code
```
  if (types_are_equal<PreconditionerType,DiagonalMatrix<VectorType> >::value == false ||
      (types_are_equal<VectorType,dealii::Vector<typename VectorType::value_type> >::value == false
       &&
       types_are_equal<VectorType,LinearAlgebra::distributed::Vector<typename VectorType::value_type> >::value == false
      ))
```
which I'm going to ignore. PVS apparently does not like template programming... ;-)

The fourth one will be addressed in #3476, the fifth I'm going to ignore again because `VectorType` does have a default constructor. (I guess the warning is again due to the template.)
